android/src/androidTest/java/bbct/android/common/navigation/test/NavigateUpFromAboutTest.java

```diff
@@ -0,0 +1,144 @@
+        context = getInstrumentation().getTargetContext();
+        this.dbUtil = new DatabaseUtil(context);
+    }
+
+    private void skipSurvey() {
+        try {
+            onView(withText(R.string.survey1)).check(matches(isDisplayed()));
+            onView(withText(R.string.later))
+                    .check(matches(isDisplayed()))
+                    .perform(click());
+        } catch (NoMatchingViewException e) {
+            //view not displayed logic
+        }
+    }
+
+    @Test
+    public void testNavigateToBaseballCardDetailsThenAboutFragment() {
+        skipSurvey();
+        String expectedTitle = context.getString(R.string.app_name);
+
+        // pop to BaseballCardList first if db is empty
+        if (this.dbUtil.isEmpty()) {
+            onView(allOf(withContentDescription(R.string.abc_action_bar_up_description), isDisplayed())).perform(click());
+        }
+        // navigate to BaseballCardDetails
+        onView(allOf(withContentDescription(R.string.add_menu), isDisplayed())).perform(click());
+
+        // navigate to About
+        openActionBarOverflowOrOptionsMenu(context);
+        String aboutTitle = context.getString(R.string.about_title);
+        onView(allOf(withText(aboutTitle), isDisplayed())).perform(click());
+
+        // Navigate Up
+        onView(allOf(withContentDescription(R.string.abc_action_bar_up_description), isDisplayed())).perform(click());
+        onView(allOf(withText(expectedTitle), isDisplayed()));
+    }
+
+    @Test
+    public void testNavigateToFilterThenAboutFragment() {
+        skipSurvey();
+        String expectedTitle = context.getString(R.string.app_name);
+
+        // pop to BaseballCardList first if db is empty
+        if (this.dbUtil.isEmpty()) {
+            onView(allOf(withContentDescription(R.string.abc_action_bar_up_description), isDisplayed())).perform(click());
+        }
+
+        // navigate to FilterCards
+        onView(allOf(withContentDescription(R.string.filter_cards_title), isDisplayed())).perform(click());
+
+        // navigate to About
+        openActionBarOverflowOrOptionsMenu(context);
+        String aboutTitle = context.getString(R.string.about_title);
+        onView(allOf(withText(aboutTitle), isDisplayed())).perform(click());
+
+        // Navigate Up
+        onView(allOf(withContentDescription(R.string.abc_action_bar_up_description), isDisplayed())).perform(click());
+        onView(allOf(withText(expectedTitle), isDisplayed()));
+    }
+
+    @Test
+    public void testNavigateToAboutFragment() {
+        skipSurvey();
+        openActionBarOverflowOrOptionsMenu(context);
+
+        String aboutTitle = context.getString(R.string.about_title);
+        onView(allOf(withText(aboutTitle), isDisplayed())).perform(click());
+
+        String expectedTitle = context.getString(R.string.app_name, aboutTitle);
+        onView(withText(expectedTitle)).check(matches(isDisplayed()));
+    }
```

Comment:
This test is really good. It Performs an action step-by-step to get an expected result. Use this pattern in all of your tests.